### PR TITLE
Jobpage

### PR DESCRIPTION
**Pull request by majakudlicka**
Created a component to display job details

### CODE REVIEW

File: react-ui/src/actions/job_apply.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import {APPLY_JOB} from '../constants/action_types.js';
+import axios from 'axios';
+
+export const applyJob = job_id => dispatch => {
+    axios
```
Comment (des-des) [inline react-ui/src/actions/job_apply.js]: You should also dispatch a 
```js
{
    type: APPLY_JOB,
    status: 'pending'
}
```

File: react-ui/src/actions/search_jobs.js [hunk 2]
```diff
@@ -35,3 +36,12 @@ export const setTerm = SearchTerm => dispatch => {
         response: SearchTerm
     });
 };
+
+export const selectJob = Job => dispatch => {
+    console.log('inside actions');
```
Comment (des-des) [inline react-ui/src/actions/search_jobs.js]: :fire:

File: react-ui/src/components/Form_Post_Job.js [hunk 3]
```diff
@@ -31,6 +31,8 @@ class Form_Post_Job extends Component {
     _onSubmit(evt) {
         evt.preventDefault();
         this.props.postJob(this.state.jobData);
+        console.log('redirecting');
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: :fire:

File: react-ui/src/components/pages/JobDetail.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            <div>
+                {this.renderJob()}
+            </div>
+        );
+    }
+}
+
+function mapStateToProps(state, ownProps) {
+    if (!state.searchJobs.jobsRequest.response) {
+        return {};
```
Comment (des-des) [inline react-ui/src/components/pages/JobDetail.js]: Not sure about this..
```js
const jobsRequestResponse = state.searchJobs.jobsRequest.response
const jobs = jobsRequestResponse && jobsRequestResponse.jobsList

return {
  applied: state.applyJob.applied,
  job_id: ownProps.params.id,
  jobs
}
```
is better

File: react-ui/src/components/pages/SearchJobs.js [hunk 5]
```diff
@@ -10,7 +11,9 @@ class BrowseJobs extends Component {
         this.formatDate = this.formatDate.bind(this);
         this.formatTime = this.formatTime.bind(this);
         this.onSubmit = this.onSubmit.bind(this);
+        this.formatDesc = this.formatDesc.bind(this);
         this.onJobSearchChange = this.onJobSearchChange.bind(this);
+        // this.onClick = this.onClick.bind(this);
```
Comment (des-des) [inline react-ui/src/components/pages/SearchJobs.js]: :fire:

File: react-ui/src/components/pages/JobDetail.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    formatDate(date) {
+        return date.slice(0, 10);
+    }
+
+    formatTime(time) {
+        return time.slice(0, 5);
+    }
+
+    sendEmail() {
+        this.props.applyJob(this.props.job_id);
```
Comment (des-des) [inline react-ui/src/components/pages/JobDetail.js]: and this `submitJobApplication`

File: react-ui/src/components/pages/JobDetail.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+    formatDate(date) {
+        return date.slice(0, 10);
+    }
+
+    formatTime(time) {
+        return time.slice(0, 5);
+    }
+
+    sendEmail() {
```
Comment (des-des) [inline react-ui/src/components/pages/JobDetail.js]: Call this `submitJobApplication`

File: react-ui/src/reducers/apply_job.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import {APPLY_JOB} from '../constants/action_types.js';
+
+const initialState = {
+    applied: []
+};
+
+export default (state = initialState, action) => {
+    switch (action.type) {
+    case APPLY_JOB:
+        return {
```
Comment (des-des) [inline react-ui/src/reducers/apply_job.js]: This is broken.

You are not storing the status, also, in the case of `status === 'error'` response in `undefined`

File: server/index.js [hunk 9]
```diff
@@ -93,21 +91,46 @@ server.register(plugins, err => {
...
+                accessKeyId: process.env.SES_ACCESS_ID,
+                secretAccessKey: process.env.SES_ACCESS_KEY,
+                region: 'eu-west-1'
+            };
+
+            var ses = new aws.SES({apiVersion: '2010-12-01'});
+            var to = ['rmrajaa@gmail.com'];
+            var from = 'maja.kudlicka@gmail.com';
+
+            ses.sendEmail(
```
Comment (des-des) [inline server/index.js]: Build a helper function (in a different file).
```js
const ses = new aws.SES ...
module.exports.sendEmail = (to, from, subject, text, cb) => {
}
```

File: server/index.js [hunk 10]
```diff
@@ -93,21 +91,46 @@ server.register(plugins, err => {
...
-                    reply.status(500)(
-                        'Failed to connect load data from the database'
-                    );
-                else {
-                    reply({
-                        name: 'jobsList',
-                        message: 'Welcome to BEEVR!',
-                        jobsList: res
-                    });
+            aws.config = {
```
Comment (des-des) [inline server/index.js]: Do you need to also store the job in the db?

File: react-ui/src/actions/job_apply.js (hunk 1 above)
Comment (majakudlicka) [inline react-ui/src/actions/job_apply.js]: ok

File: react-ui/src/components/pages/JobDetail.js (hunk 7 above)
Comment (majakudlicka) [inline react-ui/src/components/pages/JobDetail.js]: ok

File: react-ui/src/components/pages/JobDetail.js (hunk 6 above)
Comment (majakudlicka) [inline react-ui/src/components/pages/JobDetail.js]: ok

File: server/index.js (hunk 10 above)
Comment (majakudlicka) [inline server/index.js]: I guess we will need to post student_id to the job database (but we dont have student_id until we implement session management)

File: server/index.js (hunk 9 above)
Comment (majakudlicka) [inline server/index.js]: OK

**Pull request merged**